### PR TITLE
tatsu: init at 4.4.0, ics: init at 0.6

### PR DESCRIPTION
**Pull request by primeos**
###### Motivation for this change

Successfully tested the `ics` library. The expression for `tatsu` is a bit hacky :o
Optional TODOs:
- Should the attribute for TatSu be case-sensitive?
- Fixing the tests (goes a bit beyond my Python knowledge...)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/tatsu/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  #   /build/source/dist/tmpbuild/TatSu/test/ast_test.py
+  # which is not the same as the test file we want to collect:
+  #   /build/source/test/ast_test.py
+  # HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
+  doCheck = false;
+
+  propagatedBuildInputs = [ colorama mypy pyyaml regex ]
+    ++ stdenv.lib.optionals (pythonOlder "3.7") [ dataclasses ]
+    ++ stdenv.lib.optionals (pythonOlder "3.5") [ typing ]
+    ++ stdenv.lib.optional (!doCheck) pytestrunner; # TODO
```
Comment (jonringer) [inline pkgs/development/python-modules/tatsu/default.nix]:
```suggestion
    nativeBuildInputs = [ pytestrunner ];
```

File: pkgs/development/python-modules/tatsu/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  checkInputs = [ pytestrunner pytest-mypy ];
```
Comment (jonringer) [inline pkgs/development/python-modules/tatsu/default.nix]: pytestrunner is a setup_reqiures package, should go to nativeBuildInputs
```suggestion
  checkInputs = [ pytestrunner pytest-mypy ];
```

File: pkgs/development/python-modules/tatsu/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  # TODO: Need to fix errors like this one:
+  # ______________________ ERROR collecting test/ast_test.py _______________________
+  # import file mismatch:
+  # imported module 'test.ast_test' has this __file__ attribute:
+  #   /build/source/dist/tmpbuild/TatSu/test/ast_test.py
+  # which is not the same as the test file we want to collect:
+  #   /build/source/test/ast_test.py
+  # HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/tatsu/default.nix]: then specify a path for pytest to collect tests
```suggestion
  checkPhase = ''
    pytest test/
  '';
```

**Comment by primeos** [comment]
@jonringer awesome, that did fix the tests :) Thank you very much for your help!

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
both have tests
executables in tatsu seem to work

```
[7 built, 6 copied (0.4 MiB), 0.1 MiB DL]
https://github.com/NixOS/nixpkgs/pull/72111
4 package were build:
python37Packages.ics python37Packages.tatsu python38Packages.ics python38Packages.tatsu
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.ics python37Packages.tatsu python38Packages.ics python38Packages.tatsu

**Pull request merged**